### PR TITLE
Added Copy-To-Clipboard for Raid Scouting

### PR DESCRIPTION
**Pull request by TimLehner**
When enabled, will copy the raid layout + puzzle/combat names to clipboard for saving during the raid in e.g. notes or similar

During raids as soon as the raid has begun the overlay disappears. This can be annoying if later trying to remember what order and what is next, particularly where scav/farm rooms are

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsConfig.java [hunk 1]
```diff
@@ -32,109 +32,122 @@
...
 {
 	@ConfigItem(
-		position = 0,
-		keyName = "raidsTimer",
-		name = "Display elapsed raid time",
-		description = "Display elapsed raid time"
+			position = 0,
+			keyName = "raidsTimer",
+			name = "Display elapsed raid time",
+			description = "Display elapsed raid time"
```
Comment (William1Gray) [inline runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsConfig.java]: Fix double indent

File: runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsConfig.java [hunk 2]
```diff
@@ -32,109 +32,122 @@
...
 
 	@ConfigItem(
-		position = 1,
-		keyName = "pointsMessage",
-		name = "Display points in chatbox after raid",
-		description = "Display a message with total points, individual points and percentage at the end of a raid"
+			position = 1,
+			keyName = "pointsMessage",
+			name = "Display points in chatbox after raid",
+			description = "Display a message with total points, individual points and percentage at the end of a raid"
```
Comment (William1Gray) [inline runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsConfig.java]: Fix double indents

File: runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsConfig.java [hunk 3]
```diff
@@ -32,109 +32,122 @@
...
 
 	@ConfigItem(
-		position = 2,
-		keyName = "scoutOverlay",
-		name = "Show scout overlay",
-		description = "Display an overlay that shows the current raid layout (when entering lobby)"
+			position = 2,
+			keyName = "scoutOverlay",
+			name = "Show scout overlay",
+			description = "Display an overlay that shows the current raid layout (when entering lobby)"
```
Comment (William1Gray) [inline runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsConfig.java]: Fix double indents

File: runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsConfig.java [hunk 4]
```diff
@@ -32,109 +32,122 @@
...
 
 	@ConfigItem(
-		position = 3,
-		keyName = "scoutOverlayAtBank",
-		name = "Show scout overlay outside lobby",
-		description = "Keep the overlay active while at the raids area"
+			position = 3,
+			keyName = "scoutOverlayAtBank",
+			name = "Show scout overlay outside lobby",
+			description = "Keep the overlay active while at the raids area"
```
Comment (William1Gray) [inline runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsConfig.java]: Fix double indents

File: runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsConfig.java [hunk 5]
```diff
@@ -32,109 +32,122 @@
...
+
 	@ConfigItem(
-		position = 4,
-		keyName = "whitelistedRooms",
-		name = "Whitelisted rooms",
-		description = "Display whitelisted rooms in green on the overlay. Separate with comma (full name)"
+			position = 5,
+			keyName = "whitelistedRooms",
+			name = "Whitelisted rooms",
+			description = "Display whitelisted rooms in green on the overlay. Separate with comma (full name)"
```
Comment (William1Gray) [inline runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsConfig.java]: Fix double indents

File: runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsConfig.java [hunk 6]
```diff
@@ -32,109 +32,122 @@
...
 
 	@ConfigItem(
-		position = 5,
-		keyName = "blacklistedRooms",
-		name = "Blacklisted rooms",
-		description = "Display blacklisted rooms in red on the overlay. Separate with comma (full name)"
+			position = 6,
+			keyName = "blacklistedRooms",
+			name = "Blacklisted rooms",
+			description = "Display blacklisted rooms in red on the overlay. Separate with comma (full name)"
```
Comment (William1Gray) [inline runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsConfig.java]: Fix double indents

File: runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsConfig.java [hunk 7]
```diff
@@ -32,109 +32,122 @@
...
 
 	@ConfigItem(
-		position = 6,
-		keyName = "enableRotationWhitelist",
-		name = "Enable rotation whitelist",
-		description = "Enable the rotation whitelist"
+			position = 7,
+			keyName = "enableRotationWhitelist",
+			name = "Enable rotation whitelist",
+			description = "Enable the rotation whitelist"
```
Comment (William1Gray) [inline runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsConfig.java]: Fix double indents

File: runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsConfig.java [hunk 8]
```diff
@@ -32,109 +32,122 @@
...
 
 	@ConfigItem(
-		position = 7,
-		keyName = "whitelistedRotations",
-		name = "Whitelisted rotations",
-		description = "Warn when boss rotation doesn't match a whitelisted one. Add rotations like [tekton, muttadile, guardians]"
+			position = 8,
+			keyName = "whitelistedRotations",
+			name = "Whitelisted rotations",
+			description = "Warn when boss rotation doesn't match a whitelisted one. Add rotations like [tekton, muttadile, guardians]"
```
Comment (William1Gray) [inline runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsConfig.java]: Fix double indents

File: runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsConfig.java [hunk 9]
```diff
@@ -32,109 +32,122 @@
...
 
 	@ConfigItem(
-		position = 8,
-		keyName = "enableLayoutWhitelist",
-		name = "Enable layout whitelist",
-		description = "Enable the layout whitelist"
+			position = 9,
+			keyName = "enableLayoutWhitelist",
+			name = "Enable layout whitelist",
+			description = "Enable the layout whitelist"
```
Comment (William1Gray) [inline runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsConfig.java]: Fix double indents

File: runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsConfig.java [hunk 10]
```diff
@@ -32,109 +32,122 @@
...
 
 	@ConfigItem(
-		position = 9,
-		keyName = "whitelistedLayouts",
-		name = "Whitelisted layouts",
-		description = "Warn when layout doesn't match a whitelisted one. Add layouts like CFSCPPCSCF separated with comma"
+			position = 10,
+			keyName = "whitelistedLayouts",
+			name = "Whitelisted layouts",
+			description = "Warn when layout doesn't match a whitelisted one. Add layouts like CFSCPPCSCF separated with comma"
```
Comment (William1Gray) [inline runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsConfig.java]: Fix double indents

File: runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsConfig.java [hunk 11]
```diff
@@ -32,109 +32,122 @@
...
 	{
 		return true;
 	}
 
+
+	@ConfigItem(
+			position = 4,
+			keyName = "copyOverlayToClipboard",
+			name = "Copy scout overlay info to clipboard to keep during raid",
+			description = "Allows for easy pasting of layout into notes during raids"
```
Comment (William1Gray) [inline runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsConfig.java]: Fix double indents

**Comment by TimLehner** [comment]
Added a Thieving plugin that keeps track of which chests still need to be opened when looking for bats, as well as which chests contain poison

image

**Comment by Nightfirecat** [comment]
Needs rebase

**Comment by deathbeam** [comment]
I think #4365 i way better way to do this

**Comment by TimLehner** [comment]
It is useful to have it copied for pasting into discord

**Pull request closed without merging**